python/tank/platform/engine.py

```diff
@@ -504,7 +505,17 @@ def execute_in_main_thread(self, func, *args, **kwargs):
         else:
             # we don't have an invoker so just call the function:
             return func(*args, **kwargs)
-                
+
+    def renew_session(self):
+        """
+        Reauthenticate with the currently logged in user.
+        """
+        if self.has_ui:
+            # FIXME: Invoke from main thread.
```

Comment:
Yes. I simply didn't want to get into multi-thread stuff right away because this was a separate ticket. I'm already doing something similar in my ui branch.